tests/spec/core/dfn-spec.js

```diff
@@ -655,5 +655,32 @@ describe("Core — Definitions", () => {
       expect(errors[0].message).toContain("Declares both");
       expect(errors[1].message).toContain("but also has a");
     });
+
+    it("assigns data-defines on well-known pattern", async () => {
+      const body = `
+        <section>
+          <h2>Definition and its description</h2>
+          <p id="desc1" class="definition">
+            A <dfn>definition</dfn> can also have an associated description
+          </p>
+          <dl class="definition">
+            <dt><dfn>different convention</dfn></dt>
+            <dd id="desc2">Different conventions can be applied to associate a term with its description</dd>
+          </dl>
+          <dl>
+            <dt class="definition"><dfn>local convention</dfn></dt>
```

Comment:
Can we expect `dt.definition` to contain anything other than `dfn`? Seems a bit redundant markup if not. I know we need to process both dt and dfn for different reasons, so this convention feels a bit unnatural..

Reply:
Thinking about it, I wonder whether it wouldn't be more intuitive for authors to write:

```html
<dt><dfn>local convention</dfn></dt>
<dd class="definition">The prose definition of the previous dfn</dd>
```

This would help capture additional cases automatically as well, such as a `<dfn>` in a heading followed by a paragraph that contains the actual definition. In other words, the code could look for the `<dfn>` within the element that has a `class="definition"` attribute, and if not found look for the closest `<dfn>` before that element.

If it's attached to the `<dt>`, a `hasdefinition` class might be clearer.

Reply:
I've updated the code to match that suggestion, which I agree seems more author-friendly and flexible.

For `<dl>/<dt>`, I have moved to having a new `hasdefinitions` class, only for `<dl>` - individual entries can be managed using the regular "definition" class.

Reply:
Yeah, this is probably the right thing to do here (as an explicit opt-in) for this new feature. 

We should eventually also cater for:

```HTML
<dl> 
<div class="definition">
  <dt><dfn>dd</dfn><dt>
  <dd>d1</dd>
  <dd>d2</dd>
</div>
</dl>
```